File: src/frontend/src/app/models/optimization-response.ts
export interface OptimizationResponse{
    recipes: RecipeResult[]
    maxPeopleFed: number;
    unusedIngredients: UnusedIngredients[];
}

export interface RecipeResult {
    name: string;
    quantity: number;
    ingredients: RecipeIngredientResult[];
}

export interface RecipeIngredientResult {
    name: string;
    quantityPerServing: number;
    totalUsed: number
}

export interface UnusedIngredients{
    name: string;
    quantity: number;
}

File: src/backend/PantryPilot.Api/Models/Responses/OptimizationResponse.cs
namespace PantryPilot.Api.Models.Responses
{
    public class OptimizationResponse
    {
        public required List<RecipeResult> Recipes { get; set; }

        public int MaxPeopleFed { get; set; }

        public required List<UnusedIngredients> UnusedIngredients { get; set; }
    }

    public class RecipeResult
    {
        public required string Name { get; set; }
        public int Quantity { get; set; }
        public required List<RecipeIngredientResult> Ingredients { get; set; }

    }

    public class RecipeIngredientResult
    {
        public required string Name { get; set; }
        public int QuantityPerServing { get; set; }
        public int TotalUsed { get; set; }
    }

    public class UnusedIngredients
    {
        public required string Name { get; set; }
        public int Quantity { get; set; }


    }
}
